sklearn/decomposition/tests/test_pca.py

```diff
@@ -529,6 +529,26 @@ def test_pca_score3():
     assert_true(ll.argmax() == 1)
 
 
+def test_pca_score4():
+    # Ensure that the scores are correctly calculated in extreme situations
+    # Specially designed for issue #7568, #8541, #8544
+    digits = datasets.load_digits()
+    X_digits = digits.data
+
+    pca1 = PCA(n_components=30, svd_solver='full')
+    pca1.fit(X_digits)
+    score1 = pca1.score(X_digits)
+    pca2 = PCA(n_components=30, svd_solver='arpack')
+    pca2.fit(X_digits)
+    score2 = pca2.score(X_digits)
+    pca3 = PCA(n_components=30, svd_solver='randomized')
```

Comment:
Use the `random_state` and set it to anything your like e.g. 0 to ensure determinism